pkg/kthena-router/controller/modelserver_controller.go

```diff
@@ -209,6 +209,15 @@ func (c *ModelServerController) syncModelServerHandler(key string) error {
 	}
 
 	_ = c.store.AddOrUpdateModelServer(ms, pods)
+
+	// Re-enqueue all matching pods to ensure they are associated with this ModelServer
+	// This handles the case where pods became ready before the ModelServer was synced
+	for _, pod := range podList {
+		if isPodReady(pod) {
+			c.enqueuePod(pod)
```

Comment:
seems you are fixing a bug. But can you please elaborate why we need to sync pod again? And maybe we need to call pod updating directly instead of enqueue again, the `syncPodHandler` is not efficient, it would need to match by labels again

Reply:
> seems you are fixing a bug. But can you please elaborate why we need to sync pod again? And maybe we need to call pod updating directly instead of enqueue again, the `syncPodHandler` is not efficient, it would need to match by labels again

If pod is ready before handling of modelserver event, the pod will never be added to store.

Also `AddOrUpdateModelServer` will not add pods to `store.pods`. But `GetPodsByModelServer` will only get pods if it exists in `store.pods`. So we will not get pods corresponding to the modelserver in this situation.

For existing E2E test case which pods take a long time to be ready. So this issue will not be triggered.